### PR TITLE
Handle GroupTagKey and GroupTagValue when merging issues

### LINKED ISSUE #2804: User count incorrect after merging events
After I manually merge events, the user count doesn't update. For example, these two screenshots are from the same event:

image

image

This is annoying because the user count is a good way to prioritize getting errors fixed.

### PR DESCRIPTION
**Pull request by dcramer**
Fixes GH-2804

This reverts commit ef9986a9563a6cc44f9a7c3aca15cc3e67dfb404.

@getsentry/api

### CODE REVIEW

File: src/sentry/tasks/merge.py [hunk 1]
```diff
@@ -179,7 +181,27 @@ def merge_objects(models, group, new_group, limit=1000,
                 delete = True
             else:
                 delete = False
+
             if delete:
+                # Before deleting, we want to merge in counts
+                try:
+                    if isinstance(model, GroupTagKey):
```
Comment (mitsuhiko) [inline src/sentry/tasks/merge.py]: I'm confused. Those two branches look the same to me.

Comment (mitsuhiko) [inline src/sentry/tasks/merge.py]: ignore me. values_seen vs times_seen.

File: tests/sentry/tasks/test_merge.py [hunk 2]
```diff
@@ -51,6 +51,29 @@ def test_merge_creates_redirect(self):
             group_id=groups[2].id,
         ).count() == 2
 
+    def test_merge_updates_tag_values_seen(self):
+        project = self.create_project()
+        groups = [self.create_group(project) for _ in xrange(0, 2)]
+
+        for group in groups:
```
Comment (mattrobenolt) [inline tests/sentry/tasks/test_merge.py]: I was going to do this, but to assert against the regression, we need different `GroupTagKey` objects with different `key`s.

**Comment by mattrobenolt** [comment]
I'm commandeering this and adding more test cases to prevent a regression here again

**Pull request merged**